### DISCUSSION TITLE
Potential Many-To-Many Relationship with other model not referencing an array

### DISCUSSION THREAD

**Discussion by nadecancode** [prisma/prisma#14728] [Q&A]
Hi community, I'm having some issues with how to model a potential many-to-many relationship. I have a schema like this ready

```prisma
model Anime {
    id                  String          @id  @default(cuid())
    relations           Relation[]

    createdAt   DateTime   @default(now())
    updatedAt   DateTime   @updatedAt

    @@map("anime")
}

model Relation {
    type        RelationType
    animeId     String
    anime       Anime      @relation(fields: [animeId], references: [id], onDelete: Cascade)

    @@id([type, animeId])
}

enum RelationType {
    PREQUEL
    SEQUEL
    PARENT
    SIDE_STORY
    OTHER
}
```
However, what I wanted to accomplish is like this: one anime can have multiple relations and each anime can possibly share one relation entry with other anime. (Similar to edges connected with each other)
I was originally using one-to-many relationship but I quickly realized that would cause unique constraints error (``→ 156 await this.databaseService.anime.update(
  Unique constraint failed on the fields: (`type`,`animeId`)``)due to the fact that there's a potential many-to-many relationship here. Does anyone know how to correctly model this? Appreciated :)

**Comment by austincrim** [comment]
Hey @NADESHIKON 👋,

Could you provide an example of the "potential many-to-many relationship"?

**Reply by nadecancode** [reply]
Yes, let's say I have an anime called "X"
That X as a side story type relation with Y
there is an another anime called Z that also has a side story relation with Y
In this case, the relation with anime Y of type side story will be many-to-many.

**Reply by pantharshit00** [reply]
Hello @NADESHIKON , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

I think the follow model should work here:
```prisma
model Anime {
  id String @id @default(cuid())

  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  relatedTo Related[] @relation("animeRelatedTo")
  relatedBy Related[] @relation("animeRelatedBy")

  @@map("anime")
}

model Related {
  animeRelatedTo   Anime  @relation(name: "animeRelatedTo", fields: [animeRelatedToId], references: [id])
  animeRelatedToId String

  animeRelatedBy   Anime  @relation(name: "animeRelatedBy", fields: [animeRelatedById], references: [id])
  animeRelatedById String

  type RelationType

  @@id([animeRelatedToId, animeRelatedById])
}

enum RelationType {
  PREQUEL
  SEQUEL
  PARENT
  SIDE_STORY
  OTHER
}
```

Here you can relate the animes together and maybe join the two list if you want them combined.

**Reply by pantharshit00** [reply]
Hello @NADESHIKON,

I was going through our old discussions again and saw you haven't confirmed if this solved it for you. Let me know if this discussion is still relevant to you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
